File: src/main/java/chloe/movietalk/dto/response/director/DirectorInfo.java
package chloe.movietalk.dto.response.director;

import chloe.movietalk.domain.Director;
import chloe.movietalk.domain.enums.Gender;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Getter
@NoArgsConstructor
public class DirectorInfo {

    @Schema(description = "감독 ID")
    private UUID id;

    @Schema(description = "감독 이름")
    private String name;

    @Schema(description = "성별")
    private Gender gender;

    @Schema(description = "국적")
    private String country;

    @Builder
    public DirectorInfo(UUID id, String name, Gender gender, String country) {
        this.id = id;
        this.name = name;
        this.gender = gender;
        this.country = country;
    }

    public static DirectorInfo fromEntity(Director director) {
        return DirectorInfo.builder()
                .id(director.getId())
                .name(director.getName())
                .gender(director.getGender())
                .country(director.getCountry())
                .build();
    }
}

File: src/main/java/chloe/movietalk/dto/response/director/DirectorInfo.kt
package chloe.movietalk.dto.response.director

import chloe.movietalk.domain.Director
import chloe.movietalk.domain.enums.Gender
import io.swagger.v3.oas.annotations.media.Schema
import java.util.*

data class DirectorInfo(
    @field:Schema(description = "감독 ID")
    val id: UUID,

    @field:Schema(description = "감독 이름")
    val name: String,

    @field:Schema(description = "성별")
    val gender: Gender,

    @field:Schema(description = "국적")
    val country: String
) {
    companion object {
        @JvmStatic
        fun fromEntity(director: Director): DirectorInfo {
            return DirectorInfo(
                id = requireNotNull(director.id) { "Director ID must not be null"},
                name = director.name,
                gender = director.gender,
                country = director.country
            )
        }
    }
}
